src/main/java/io/wispforest/owo/network/OwoHandshake.java

```diff
@@ -110,49 +111,39 @@ private static void configureStart(ServerConfigurationNetworkHandler handler, Mi
             return;
         }
 
-        var request = PacketByteBufs.create();
-        writeHashes(request, OwoNetChannel.OPTIONAL_CHANNELS, OwoHandshake::hashChannel);
-        ServerConfigurationNetworking.send(handler, OwoHandshake.CHANNEL_ID, request);
+        var optionalChannels = formatHashes(OwoNetChannel.OPTIONAL_CHANNELS, OwoHandshake::hashChannel);
+        ServerConfigurationNetworking.send(handler, new HandshakeRequest(optionalChannels));
         Owo.LOGGER.info("[Handshake] Sending channel packet");
     }
 
     @Environment(EnvType.CLIENT)
-    private static void syncClient(MinecraftClient client, ClientConfigurationNetworkHandler handler, PacketByteBuf buf, PacketSender sender) {
+    private static void syncClient(HandshakeRequest request, ClientConfigurationNetworking.Context context) {
         Owo.LOGGER.info("[Handshake] Sending client channels");
         QUERY_RECEIVED = true;
 
-        if (buf.readableBytes() > 0) {
-            final var serverOptionalChannels = buf.read(CHANNEL_HASHES_ENDEC);
-            ((OwoClientConnectionExtension) ((ClientCommonNetworkHandlerAccessor) handler).getConnection()).owo$setChannelSet(filterOptionalServices(serverOptionalChannels, OwoNetChannel.REGISTERED_CHANNELS, OwoHandshake::hashChannel));
-        }
+        // TODO: get the actual network handler here.
```

Comment:
What's the deal with this exactly? Since uncommenting this line at the very least doesn't cause a compile error I'd assume there is some deeper reason behind this comment